src/games/strategy/triplea/ai/proAI/util/ProUtils.java

```diff
@@ -27,15 +27,10 @@
  */
 public class ProUtils {
 
-  public static Map<Unit, Territory> createUnitTerritoryMap(final PlayerID player) {
-    final GameData data = ProData.getData();
-    final List<Territory> allTerritories = data.getMap().getTerritories();
-    final List<Territory> myUnitTerritories =
-        Match.getMatches(allTerritories, Matches.territoryHasUnitsOwnedBy(player));
+  public static Map<Unit, Territory> createUnitTerritoryMap() {
     final Map<Unit, Territory> unitTerritoryMap = new HashMap<Unit, Territory>();
-    for (final Territory t : myUnitTerritories) {
-      final List<Unit> myUnits = t.getUnits().getMatches(Matches.unitIsOwnedBy(player));
-      for (final Unit u : myUnits) {
+    for (final Territory t : ProData.getData().getMap().getTerritories()) {
```

Comment:
Looks like there are a few NPE potential spots here. Also,  the onner loop could be replaced with a Lambda expression to make this a bit easier to read.